conda_index/cli/__init__.py

```diff
@@ -115,7 +115,7 @@
     help="""
         Skip generating current_repodata.json, a file containing only the newest
         versions of all packages and their dependencies, only used by the
-        classic solver.
+        classic solver. Only works with --write-monolithic.
```

Comment:
```suggestion
        classic solver.  Conflicts with --no-write-monolithic
```

Or "Incompatible with --no-write-monolithic" perhaps. Or doesn't make sense when --write-monolithic is not passed?

Since `--write-monolithic` is the default we don't need to ask the user to enable it specifically.